core/domain/collection_domain.py

```diff
@@ -32,50 +33,69 @@
+COLLECTION_NODE_PROPERTY_ACQUIRED_SKILLS: Final = 'acquired_skills'
 
 # This takes additional 'title' and 'category' parameters.
-CMD_CREATE_NEW = 'create_new'
+CMD_CREATE_NEW: Final = 'create_new'
 # This takes an additional 'exploration_id' parameter.
-CMD_ADD_COLLECTION_NODE = 'add_collection_node'
+CMD_ADD_COLLECTION_NODE: Final = 'add_collection_node'
 # This takes an additional 'exploration_id' parameter.
-CMD_DELETE_COLLECTION_NODE = 'delete_collection_node'
+CMD_DELETE_COLLECTION_NODE: Final = 'delete_collection_node'
 # This takes 2 parameters corresponding to the node indices to be swapped.
-CMD_SWAP_COLLECTION_NODES = 'swap_nodes'
+CMD_SWAP_COLLECTION_NODES: Final = 'swap_nodes'
 # This takes additional 'property_name' and 'new_value' parameters and,
 # optionally, 'old_value'.
-CMD_EDIT_COLLECTION_PROPERTY = 'edit_collection_property'
+CMD_EDIT_COLLECTION_PROPERTY: Final = 'edit_collection_property'
 # This takes additional 'property_name' and 'new_value' parameters and,
 # optionally, 'old_value'.
 # Currently, a node only has exploration_id as its parameter, if more
 # parameters are to be added, the following CMD should be used
 # for its changelists.
-CMD_EDIT_COLLECTION_NODE_PROPERTY = 'edit_collection_node_property'
+CMD_EDIT_COLLECTION_NODE_PROPERTY: Final = 'edit_collection_node_property'
 # This takes additional 'from_version' and 'to_version' parameters for logging.
-CMD_MIGRATE_SCHEMA_TO_LATEST_VERSION = 'migrate_schema_to_latest_version'
+CMD_MIGRATE_SCHEMA_TO_LATEST_VERSION: Final = 'migrate_schema_to_latest_version'
 # This takes an additional 'name' parameter.
-CMD_ADD_COLLECTION_SKILL = 'add_collection_skill'
+CMD_ADD_COLLECTION_SKILL: Final = 'add_collection_skill'
 # This takes an additional 'skill_id' parameter.
-CMD_DELETE_COLLECTION_SKILL = 'delete_collection_skill'
+CMD_DELETE_COLLECTION_SKILL: Final = 'delete_collection_skill'
 # This takes additional 'question_id' and 'skill_id' parameters.
-CMD_ADD_QUESTION_ID_TO_SKILL = 'add_question_id_to_skill'
+CMD_ADD_QUESTION_ID_TO_SKILL: Final = 'add_question_id_to_skill'
 # This takes additional 'question_id' and 'skill_id' parameters.
-CMD_REMOVE_QUESTION_ID_FROM_SKILL = 'remove_question_id_from_skill'
+CMD_REMOVE_QUESTION_ID_FROM_SKILL: Final = 'remove_question_id_from_skill'
 
 # Prefix for skill IDs. This should not be changed -- doing so will result in
 # backwards-compatibility issues.
-_SKILL_ID_PREFIX = 'skill'
+_SKILL_ID_PREFIX: Final = 'skill'
+
+CollectionPropertiesTuple = Tuple[
+    Literal['title'], Literal['category'],
+    Literal['objective'], Literal['language_code'],
+    Literal['tags']]
+
+
+class AllowedCommandDict(TypedDict, total=False):
+    """Type for the ALLOWED COMMAND Dict"""
+
+    name: str
+    required_attribute_names: List[str]
+    optional_attribute_names: List[str]
+    user_id_attribute_names: List[str]
+    allowed_values: Dict[str, CollectionPropertiesTuple]
```

Comment:
can you please skip this for now, if possible ??
i'll correct it by tuesday morning, because i'm still figuring it out.
In your mentioned PR and my PR, approach is same but context is different. Hope it make sense!

Reply:
What should I skip? Why is the context different?

Reply:
Previously i was thinking about changing the tuple to list.
but now it is fixed.
done!!